model/test_results.go

```diff
@@ -289,6 +278,38 @@ func (t *TestResults) Close(ctx context.Context) error {
 	return errors.Wrapf(err, "problem closing test result record with id %s", t.ID)
 }
 
+func (t *TestResults) getBucket(ctx context.Context) (pail.Bucket, string, error) {
+	if t.bucket == "" {
+		conf := &CedarConfig{}
+		conf.Setup(t.env)
+		if err := conf.Find(); err != nil {
+			return nil, "", errors.Wrap(err, "problem getting application configuration")
+		}
+		t.bucket = conf.Bucket.TestResultsBucket
+	}
+
+	var prefix string
+	if t.Artifact.Type == PailLocal {
+		prefix = filepath.Join(testResultsCollection, t.Artifact.Prefix)
+	} else {
+		prefix = fmt.Sprintf("%s/%s", testResultsCollection, t.Artifact.Prefix)
+	}
+
+	bucket, err := t.Artifact.Type.Create(
+		ctx,
+		t.env,
+		t.bucket,
+		prefix,
+		string(pail.S3PermissionsPrivate),
+		true,
+	)
+	if err != nil {
+		return nil, "", errors.Wrap(err, "problem creating bucket")
+	}
+
+	return bucket, prefix, nil
```

Comment:
why do we need to return a prefix if its baked into the bucket?

Reply:
for logging I want to get the right prefix